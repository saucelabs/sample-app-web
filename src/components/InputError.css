.form_group {
  margin-bottom: 15px;
  position: relative;
}
.input_error {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  font-size: 18px;
  width: 100%;
  border: 0px;
  border-bottom: 2px solid #ededef;
  outline: none;
  padding: 10px 0;
}

.input_error.error {
  border-bottom-color: #e2231a;
}

.error_icon {
  color: #e2231a;
  font-size: 18px;
  position: absolute;
  right: 0;
  top: 12px;
}

/* Chrome/Opera/Safari */
.form_input::-webkit-input-placeholder,
    /* Firefox 19+ */
.form_input::-moz-placeholder,
    /* IE 10+ */
.form_input:-ms-input-placeholder,
.form_input::placeholder {
  color: #c1c7c6;
}

/* Chrome/Opera/Safari */
.form_input.error::-webkit-input-placeholder,
    /* Firefox 19+ */
.form_input.error::-moz-placeholder,
    /* IE 10+ */
.form_input.error:-ms-input-placeholder,
.form_input.error::placeholder {
  color: #e2231a !important;
}
