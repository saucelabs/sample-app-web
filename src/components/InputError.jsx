import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
import "./InputError.css";

export const INPUT_TYPES = {
  TEXT: "text",
  PASSWORD: "password",
};
const InputError = ({
  isError,
  onChange,
  placeholder,
  testId,
  type,
  value,
  ...props
}) => {
  return (
    <div className="form_group">
      <input
        // `form_input` has no style function
        // but is there for backwards compatibility
        className={`input_error form_input${isError ? " error" : ""}`}
        placeholder={placeholder}
        onChange={onChange}
        type={type}
        value={value}
        {...(testId
          ? {
              "data-test": testId,
              id: testId,
              name: testId,
            }
          : {})}
        {...props}
      />
      {isError && (
        <FontAwesomeIcon icon={faTimesCircle} className="error_icon" />
      )}
    </div>
  );
};

InputError.propTypes = {
  /**
   * If this is an isError field yes or no
   */
  isError: PropTypes.bool.isRequired,
  /**
   * The on change handler
   */
  onChange: PropTypes.func.isRequired,
  /**
   * The placeholder of the input
   */
  placeholder: PropTypes.string,
  /**
   * The test id
   */
  testId: PropTypes.string,
  /**
   * What type of field is it
   */
  type: PropTypes.oneOf(["text", "password"]),
  /**
   * The value of the input
   */
  value: PropTypes.string,
};

InputError.defaultProps = {
  placeholder: "",
  testId: undefined,
  type: INPUT_TYPES.TEXT,
  value: "",
};

export default InputError;
