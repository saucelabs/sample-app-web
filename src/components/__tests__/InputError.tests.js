import React from "react";
import { shallow } from "enzyme";
import InputError, { INPUT_TYPES } from "../InputError";

let props;

describe("InputError", () => {
  beforeEach(() => {
    props = {
      isError: false,
      onChange: jest.fn(),
    };
  });

  it("should render correctly with the required options", () => {
    const component = shallow(<InputError {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render secure input", () => {
    const component = shallow(
      <InputError {...props} type={INPUT_TYPES.PASSWORD} />
    );

    expect(component).toMatchSnapshot();
  });

  it("should render an error input", () => {
    const component = shallow(<InputError {...props} isError />);

    expect(component).toMatchSnapshot();
  });

  it("should render an input with a testId", () => {
    const component = shallow(<InputError {...props} testId="some-id" />);

    expect(component).toMatchSnapshot();
  });

  it("should render an input with custom props", () => {
    const component = shallow(
      <InputError {...props} testId="some-id" name="own-name-id" foo="bar" />
    );

    expect(component).toMatchSnapshot();
  });

  it("should render an input with full props", () => {
    const component = shallow(
      <InputError
        {...props}
        testId="some-id"
        name="own-name-id"
        foo="bar"
        placeholder="The placeholder"
        value="the value"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
