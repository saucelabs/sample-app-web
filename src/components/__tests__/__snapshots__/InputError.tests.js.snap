// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError should render an error input 1`] = `
<div
  className="form_group"
>
  <input
    className="input_error form_input error"
    onChange={[MockFunction]}
    placeholder=""
    type="text"
    value=""
  />
  <FontAwesomeIcon
    border={false}
    className="error_icon"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f057",
          "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
        ],
        "iconName": "times-circle",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;

exports[`InputError should render an input with a testId 1`] = `
<div
  className="form_group"
>
  <input
    className="input_error form_input"
    data-test="some-id"
    id="some-id"
    name="some-id"
    onChange={[MockFunction]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`InputError should render an input with custom props 1`] = `
<div
  className="form_group"
>
  <input
    className="input_error form_input"
    data-test="some-id"
    foo="bar"
    id="some-id"
    name="own-name-id"
    onChange={[MockFunction]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`InputError should render an input with full props 1`] = `
<div
  className="form_group"
>
  <input
    className="input_error form_input"
    data-test="some-id"
    foo="bar"
    id="some-id"
    name="own-name-id"
    onChange={[MockFunction]}
    placeholder="The placeholder"
    type="text"
    value="the value"
  />
</div>
`;

exports[`InputError should render correctly with the required options 1`] = `
<div
  className="form_group"
>
  <input
    className="input_error form_input"
    onChange={[MockFunction]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`InputError should render secure input 1`] = `
<div
  className="form_group"
>
  <input
    className="input_error form_input"
    onChange={[MockFunction]}
    placeholder=""
    type="password"
    value=""
  />
</div>
`;
